<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>480</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="minimumSize">
   <size>
    <width>600</width>
    <height>480</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>600</width>
    <height>480</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>QSTLink2</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QGroupBox" name="gb_top">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>0</y>
      <width>451</width>
      <height>41</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QPushButton" name="b_connect">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Connect</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b_send">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Send...</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b_receive">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Receive...</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b_disconnect">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Disconnect</string>
     </property>
    </widget>
   </widget>
   <widget class="QPlainTextEdit" name="t_log">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>50</y>
      <width>581</width>
      <height>261</height>
     </rect>
    </property>
    <property name="font">
     <font>
      <family>Courier New</family>
      <pointsize>10</pointsize>
      <weight>50</weight>
      <bold>false</bold>
     </font>
    </property>
    <property name="undoRedoEnabled">
     <bool>false</bool>
    </property>
    <property name="readOnly">
     <bool>true</bool>
    </property>
   </widget>
   <widget class="QPushButton" name="b_quit">
    <property name="geometry">
     <rect>
      <x>510</x>
      <y>0</y>
      <width>81</width>
      <height>50</height>
     </rect>
    </property>
    <property name="text">
     <string>Quit</string>
    </property>
   </widget>
   <widget class="QGroupBox" name="gb_bottom">
    <property name="enabled">
     <bool>false</bool>
    </property>
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>410</y>
      <width>581</width>
      <height>41</height>
     </rect>
    </property>
    <property name="title">
     <string/>
    </property>
    <widget class="QPushButton" name="b_halt">
     <property name="geometry">
      <rect>
       <x>10</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Halt MCU</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b_run">
     <property name="geometry">
      <rect>
       <x>120</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Run MCU</string>
     </property>
    </widget>
    <widget class="QPushButton" name="b_reset">
     <property name="geometry">
      <rect>
       <x>230</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Reset MCU</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="r_jtag">
     <property name="geometry">
      <rect>
       <x>460</x>
       <y>10</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>JTAG</string>
     </property>
    </widget>
    <widget class="QRadioButton" name="r_swd">
     <property name="geometry">
      <rect>
       <x>520</x>
       <y>10</y>
       <width>60</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>SWD</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
    <widget class="QPushButton" name="b_getMCU">
     <property name="geometry">
      <rect>
       <x>340</x>
       <y>10</y>
       <width>100</width>
       <height>30</height>
      </rect>
     </property>
     <property name="text">
      <string>Get Flash Info</string>
     </property>
    </widget>
   </widget>
   <widget class="QTabWidget" name="tabw_info">
    <property name="geometry">
     <rect>
      <x>10</x>
      <y>310</y>
      <width>581</width>
      <height>91</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tab_mcu">
     <attribute name="title">
      <string>MCU Info</string>
     </attribute>
     <widget class="QLabel" name="l_type">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>MCU Type:</string>
      </property>
     </widget>
     <widget class="QLabel" name="l_chipid">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Chip ID:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_type">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>10</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_chipid">
      <property name="geometry">
       <rect>
        <x>70</x>
        <y>30</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="l_flashbase">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>10</y>
        <width>71</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Flash base:</string>
      </property>
     </widget>
     <widget class="QLabel" name="l_flashsize">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>30</y>
        <width>71</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Flash size:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_flashbase">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>10</y>
        <width>111</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_flashsize">
      <property name="geometry">
       <rect>
        <x>260</x>
        <y>30</y>
        <width>111</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLabel" name="l_ramsize">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>30</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Ram size:</string>
      </property>
     </widget>
     <widget class="QLabel" name="l_rambase">
      <property name="geometry">
       <rect>
        <x>390</x>
        <y>10</y>
        <width>61</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Ram base:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_ramsize">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>30</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_rambase">
      <property name="geometry">
       <rect>
        <x>450</x>
        <y>10</y>
        <width>113</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_stlink">
     <attribute name="title">
      <string>ST-Link Info</string>
     </attribute>
     <widget class="QLabel" name="l_stlver">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>20</y>
        <width>56</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Version:</string>
      </property>
     </widget>
     <widget class="QLabel" name="l_jtagver">
      <property name="geometry">
       <rect>
        <x>210</x>
        <y>20</y>
        <width>81</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>JTAG Version:</string>
      </property>
     </widget>
     <widget class="QLabel" name="l_swimver">
      <property name="geometry">
       <rect>
        <x>400</x>
        <y>20</y>
        <width>91</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>SWIM Version:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_stlver">
      <property name="geometry">
       <rect>
        <x>90</x>
        <y>20</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_jtagver">
      <property name="geometry">
       <rect>
        <x>290</x>
        <y>20</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_swimver">
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>20</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
      <property name="readOnly">
       <bool>true</bool>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_flash">
     <attribute name="title">
      <string>Flash Info</string>
     </attribute>
     <widget class="QLabel" name="l_readpr">
      <property name="geometry">
       <rect>
        <x>100</x>
        <y>20</y>
        <width>111</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Read Protection:</string>
      </property>
     </widget>
     <widget class="QLabel" name="l_writepr">
      <property name="geometry">
       <rect>
        <x>320</x>
        <y>20</y>
        <width>101</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Write Protection:</string>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_readpr">
      <property name="geometry">
       <rect>
        <x>200</x>
        <y>20</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLineEdit" name="le_writepr">
      <property name="geometry">
       <rect>
        <x>420</x>
        <y>20</y>
        <width>51</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab_progress">
     <attribute name="title">
      <string>Transfer Progress</string>
     </attribute>
     <widget class="QProgressBar" name="pgb_transfer">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>471</width>
        <height>28</height>
       </rect>
      </property>
      <property name="value">
       <number>0</number>
      </property>
     </widget>
     <widget class="QLabel" name="l_progress">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>0</y>
        <width>551</width>
        <height>30</height>
       </rect>
      </property>
      <property name="text">
       <string>No transfer in progress</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignCenter</set>
      </property>
     </widget>
     <widget class="QPushButton" name="b_stop">
      <property name="enabled">
       <bool>true</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>490</x>
        <y>30</y>
        <width>80</width>
        <height>28</height>
       </rect>
      </property>
      <property name="text">
       <string>Stop</string>
      </property>
     </widget>
    </widget>
   </widget>
   <widget class="QPushButton" name="b_help">
    <property name="geometry">
     <rect>
      <x>470</x>
      <y>0</y>
      <width>41</width>
      <height>51</height>
     </rect>
    </property>
    <property name="text">
     <string/>
    </property>
    <property name="icon">
     <iconset resource="res/ressources.qrc">
      <normaloff>:/images/icon-questionmark.png</normaloff>:/images/icon-questionmark.png</iconset>
    </property>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources>
  <include location="res/ressources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>b_quit</sender>
   <signal>clicked()</signal>
   <receiver>MainWindow</receiver>
   <slot>close()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>53</y>
    </hint>
    <hint type="destinationlabel">
     <x>299</x>
     <y>239</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>b_quit</sender>
   <signal>clicked()</signal>
   <receiver>b_stop</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>529</x>
     <y>32</y>
    </hint>
    <hint type="destinationlabel">
     <x>543</x>
     <y>386</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
